video/out/wayland_common.c

```diff
@@ -2648,6 +2729,22 @@ static void set_color_management(struct vo_wayland_state *wl)
 #endif
 }
 
+static void set_color_representation(struct vo_wayland_state *wl)
+{
+    struct mp_image_params target_params = vo_get_target_params(wl->vo);
+    if (!wl->color_representation_surface || pl_color_repr_equal(&target_params.repr, &wl->target_params.repr))
+        return;
```

Comment:
`set_color_management` always does `wl->target_params = target_params;`, so `pl_color_repr_equal(&target_params.repr, &wl->target_params.repr)` is always true

Reply:
Good point. I should probably extract out the wl->target_params handling into `vo_wayland_handle_color`.